### ISSUE TITLE
MapboxCoreNavigation fails to build on CI — Multiple commands produce MapboxCoreNavigation.swiftmodule

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#2866]
As of #2863, the Carthage-based MapboxCoreNavigation build is [failing](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/2778/workflows/c3e405b9-70c0-4ec4-95ff-03d213ead388/jobs/28199) on the release-v2.0 branch with lots of errors like this:

```
error: Multiple commands produce '/Users/distiller/Library/Developer/Xcode/DerivedData/MapboxNavigation-gtrukiuwvhbmctfcsthejciepfao/Build/Intermediates.noindex/MapboxNavigation.build/Debug-iphonesimulator/MapboxCoreNavigation.build/Objects-normal/x86_64/MapboxCoreNavigation.swiftmodule':
1) Target 'MapboxCoreNavigation' (project 'MapboxNavigation') has compile command for Swift source files
2) Target 'MapboxCoreNavigation' (project 'MapboxNavigation') has compile command for Swift source files
```

I think it might have something to do with the fact that Xcode automatically generates a scheme with the same “MapboxCoreNavigation” name as the shared scheme in the Xcode project. But the error message seems to indicate that it found the right scheme and target – just that the target has been duplicated for some reason. I haven’t managed to reproduce this issue locally in Xcode 12.4, so I don’t know if it’s specific to the rest of the CircleCI configuration.

/cc @mapbox/navigation-ios

**Comment by ShanMa1991** [comment]
I think it may be caused by the "Example" in the dependencies of the `MapboxCoreNavigation` building phase for test. Because the "Example" has the link to `MapboxNavigation`, while  `MapboxNavigation` has the dependency on `MapboxCoreNavigation`. So when building the test for the `MapboxCoreNavigation`, it would cause the duplication.

**Comment by zugaldia** [comment]
> the Carthage-based MapboxCoreNavigation build is failing

Instead of fixing this build issue, could we use this as an opportunity to remove Carthage support and switch to Cocoapods or SPM moving forward?

**Comment by 1ec5** [comment]
> Instead of fixing this build issue, could we use this as an opportunity to remove Carthage support and switch to Cocoapods or SPM moving forward?

We’re so close to making Carthage work that I’d want to leave the possibility there rather than make it much harder to support it in the future if we need to. Basically it’s a CI-only issue at this point. Note that this issue only applies to MapboxCoreNavigation; MapboxNavigation is outside the scope of this issue.

**Comment by 1ec5** [comment]
Also, as @ShanMa1991 and I are looking into these errors, they don’t appear to be completely specific to Carthage. We’re seeing some real project integrity issues that have been cropping up as we’ve ignored the failing CI job over the past few weeks.

**Comment by zugaldia** [comment]
> We’re so close to making Carthage work

![](https://media1.giphy.com/media/119SFXjoMsax6o/200.gif?cid=5a38a5a2qbwj9fnio9th4cx7gie7u0ktaq94ypidc6bie18i&rid=200.gif)

**Comment by 1ec5** [comment]
Fixed in mapbox/mapbox-navigation-ios#2874.
